src/completion.ts

```diff
@@ -157,11 +157,32 @@ class MdCompletionItemProvider implements CompletionItemProvider {
             let startIndex = lineTextBefore.lastIndexOf('[');
             const range = new Range(position.with({ character: startIndex + 1 }), position);
             return new Promise((res, _) => {
-                let refLabels = document.getText().split(/\r?\n/).reduce((prev, curr) => {
+                const lines = document.getText().split(/\r?\n/);
+                const usageCounts = lines.reduce((useCounts, currentLine) => {
+                    let match: RegExpExecArray;
+                    const pattern = /\[[^\]]+\]\[([^\]]*?)\]/g;
+                    while ((match = pattern.exec(currentLine)) !== null) {
+                        let usedRef = match[1];
+                        if (!useCounts.has(usedRef)) {
+                            useCounts.set(usedRef, 0);
+                        }
+                        useCounts.set(usedRef, useCounts.get(usedRef) + 1);
+                    }
+                    return useCounts;
+                }, new Map<string, number>());
+                let refLabels = lines.reduce((prev, curr) => {
                     let match;
                     if ((match = /^\[([^\]]*?)\]: (\S*)( .*)?/.exec(curr)) !== null) {
-                        let item = new CompletionItem(match[1], CompletionItemKind.Reference);
-                        item.documentation = match[2];
+                        const ref = match[1];
+                        let item = new CompletionItem(ref, CompletionItemKind.Reference);
+                        const usages = usageCounts.get(ref) || 0;
+                        item.documentation = new MarkdownString(match[2]);
+                        item.detail = usages === 1 ? `1 usage`  : `${usages} usages`;
+                        // prefer unused items
+                        // We need to `sortText` in the else case as well due to:
+                        // https://github.com/Microsoft/vscode/issues/66109#issuecomment-451873316
+                        item.sortText = usages === 0 ? `!!!-${ref}` : item.sortText = ref;
```

Comment:
Why not using

```suggestion
                        item.sortText = usages + ref;
```

Reply:
I don't think ordering by the count of usages in general is very useful.
I'm not more likely to want to choose a link that I've used 3 times over one that I've used 4 times.
With the current approach, labels that are used at least once will still sort the same way they did before this patch.

Reply:
Get it. Maybe `0${ref}` for 0 and `1${ref}` for >0? (I just feel `!!!-` a bit strange 😂)

(And 0 and 1 are also used to sort the KaTeX functions)

https://github.com/yzhang-gh/vscode-markdown/blob/5446a9b10e98b012eb19ff03ac15b4dfc364e3fa/src/completion.ts#L75-L84

Reply:
Changed. I have added a `-` in case VSC recognizes numbers in `sortText`.